x-pack/plugins/reporting/server/lib/validate_max_content_length.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import numeral from '@elastic/numeral';
+import { get } from 'lodash';
+
+const KIBANA_MAX_SIZE_BYTES_PATH = 'xpack.reporting.csv.maxSizeBytes';
+const ES_MAX_SIZE_BYTES_PATH = 'http.max_content_length';
+
+export async function validateMaxContentLength(
+  config: any,
+  esClient: any,
+  log: (message: string) => any
+) {
+  const kibanaMaxContentBytes = config.get(KIBANA_MAX_SIZE_BYTES_PATH);
+  const elasticClusterSettings = await esClient.cluster.getSettings({ includeDefaults: true });
+  const elasticSearchMaxContent = get(elasticClusterSettings, 'http.max_content_length', '100mb');
+  const elasticSearchMaxContentBytes = numeral().unformat(elasticSearchMaxContent.toUpperCase());
+
+  if (kibanaMaxContentBytes > elasticSearchMaxContentBytes) {
+    log(
+      `${KIBANA_MAX_SIZE_BYTES_PATH} (${kibanaMaxContentBytes}) is higher than ElasticSearch's ${ES_MAX_SIZE_BYTES_PATH} (${elasticSearchMaxContentBytes}). ` +
+        `. Setting ${KIBANA_MAX_SIZE_BYTES_PATH} to ${elasticSearchMaxContentBytes}. ` +
+        `Please set ${ES_MAX_SIZE_BYTES_PATH} in ElasticSearch to match, or lower your ${KIBANA_MAX_SIZE_BYTES_PATH} in Kibana to avoid this warning.`
+    );
+    config.set(KIBANA_MAX_SIZE_BYTES_PATH, elasticSearchMaxContentBytes);
```

Comment:
Let's not try to fix the problem for them. The Elasticsearch setting is not the only determiner on the payload size going over traffic. If they have an HTTP Proxy in front of Elasticsearch, that will have it's own setting for body size, such as `client_max_body_size` in Nginx.

I think it's enough to run the check and log a warning on startup, and if something does go wrong in report generation, log the actual error that came through, which should be self-explanatory.

Reply:
For sure. I went back and forth on this as well, but does feel too magical